framework/src/org/checkerframework/framework/type/TypeFromExpressionVisitor.java

```diff
@@ -273,13 +273,17 @@ private void annotateArrayAsArray(AnnotatedArrayType result, NewArrayTree node,
     }
 
     /**
-     * The type of a NewClassTree is the type of the Identifier
-     * plus any explicit annotations (including polymorphic qualifiers)
-     * on the constructor.
+     * Creates an AnnotatedDeclaredType for the NewClassTree and adds one of:
```

Comment:
For your example, `@A@B@C` would all be added.  I changed the documentation to clarify this.